getcontext/tracing/trace.py

```diff
@@ -93,9 +98,59 @@ def evaluate(self):
             enforce_https=enforce_https(),
         )
 
-        return self.context_client.evaluations.result(
-            id=run_details.data.run_id, enforce_https=enforce_https()
+        results = self._poll_for_result(run_details.data.run_id)
+
+        if results.status == "failed":
+            raise EvaluationError(f"Evaluation failed:\n {results.as_dict()}")
+
+        failed_evaluation_msgs = self._parse_evaluation_results(results)
+        failed_msg = self._create_evaluation_fail_msg(failed_evaluation_msgs)
+
+        if failed_msg:
+            raise EvaluationException(failed_msg)
+
+        return results
+
+    def _parse_evaluation_results(self, results: EvaluationsRunResponse):
+        outcome_map = {
+            "positive": "passed",
+            "negative": "failed",
+            "partially_passed": "partially passed",
+            "inconclusive": "inconclusive",
+        }
+
+        failed_evaluation_msgs = []
+
+        for result in results.results:
+            test_case_name = result.test_case.name
+            for evaluation in result.evaluations:
+                tick = u'\u2705'
+                x_mark = u'\u274C'
+                symbol = tick if evaluation.outcome == "positive" else x_mark
+                msg = (
+                    f"{symbol} {test_case_name}: Evaluation {outcome_map.get(evaluation.outcome)} for"
+                    f" {evaluation.evaluator_name}:\n\t {evaluation.reasoning}"
+                )
+                if evaluation.outcome == "positive":
+                    print(f"\n{msg}")
+                else:
+                    failed_evaluation_msgs.append(msg)
+        return failed_evaluation_msgs
+
+    def _create_evaluation_fail_msg(self, failed_evaluation_msgs):
+        if failed_evaluation_msgs:
+            joined_failed_evaluations = "\n\n".join(failed_evaluation_msgs)
+            return f"\n\nEvaluation failed:\n {joined_failed_evaluations}"
+
+    def _poll_for_result(self, run_id: str) -> EvaluationsRunResponse:
+        result = self.context_client.evaluations.result(
+            id=run_id, enforce_https=enforce_https()
         )
+        while result.status in ["running", "pending"]:
+            result = self.context_client.evaluations.result(
+                id=run_id, enforce_https=enforce_https()
```

Comment:
enforce_https could also be a private member on the instance given that we are reusing it in several places?